cryptofeed/exchange/binance_futures.py

```diff
@@ -115,7 +115,7 @@ async def message_handler(self, msg: str, conn: AsyncConnection, timestamp: floa
         elif msg_type == 'forceOrder':
             await self._liquidations(msg, timestamp)
         elif msg_type == 'markPriceUpdate':
-            await self._funding(msg, timestamp)
+            await self._funding_with_index_price(msg, timestamp)    # Same channel for two updates
```

Comment:
not ideal that these share a channel, means that the code gets a bit messy as you saw above.